apps/agreement/contracts/Agreement.sol

```diff
@@ -0,0 +1,1350 @@
+    )
+        external
+        auth(MANAGE_DISPUTABLE_ROLE)
+    {
+        DisputableInfo storage disputableInfo = disputableInfos[address(_disputable)];
+        _ensureRegisteredDisputable(disputableInfo);
+
+        _changeCollateralRequirement(_disputable, disputableInfo, _collateralToken, _actionAmount, _challengeAmount, _challengeDuration);
+    }
+
+    /**
+    * @notice Update Agreement to title "`_title`" and content "`_content`", with arbitrator `_arbitrator`
+    * @param _title String indicating a short description
+    * @param _content Link to a human-readable text that describes the rules for the Agreements instance
+    * @param _arbitrator Address of the IArbitrator that will be used to resolve disputes
+    */
+    function changeSetting(string _title, bytes _content, IArbitrator _arbitrator) external auth(CHANGE_AGREEMENT_ROLE) {
+        _newSetting(_title, _content, _arbitrator);
+    }
+
+    /**
+    * @notice Sign the agreement
+    */
+    function sign() external {
+        uint256 currentSettingId = _getCurrentSettingId();
+        uint256 lastSettingIdSigned = lastSettingSignedBy[msg.sender];
+        require(lastSettingIdSigned < currentSettingId, ERROR_SIGNER_ALREADY_SIGNED);
+
+        lastSettingSignedBy[msg.sender] = currentSettingId;
+        emit Signed(msg.sender, currentSettingId);
+    }
+
+    /**
+    * @notice Register action #`_disputableActionId` from disputable `msg.sender` for submitter `_submitter` with context `_context`
+    * @dev This function should be called from disputable apps every time a new disputable action is created in the app.
+    *      Each disputable action ID must only be registered once; this is how the Agreement gets notified about each potentially disputable action.
+    * @param _disputableActionId Identification number of the disputable action in the context of the disputable instance
+    * @param _submitter Address of the user that has submitted the action
+    * @param _context Link to a human-readable text providing context for the given action
+    * @return Unique identification number for the created action in the context of the agreement
+    */
+    function newAction(uint256 _disputableActionId, bytes _context, address _submitter) external returns (uint256) {
+        uint256 lastSettingIdSigned = lastSettingSignedBy[_submitter];
+        require(lastSettingIdSigned >= _getCurrentSettingId(), ERROR_SIGNER_MUST_SIGN);
+
+        DisputableInfo storage disputableInfo = disputableInfos[msg.sender];
+        _ensureRegisteredDisputable(disputableInfo);
+
+        uint256 currentCollateralRequirementId = disputableInfo.collateralRequirementsLength.sub(1);
+        CollateralRequirement storage requirement = disputableInfo.collateralRequirements[currentCollateralRequirementId];
+        _lockBalance(requirement.staking, _submitter, requirement.actionAmount);
+
+        uint256 id = actionsLength++;
+        Action storage action = actions[id];
```

Comment:
Can do last if we have room, but we have typically errored if the struct did not exist.

Reply:
Not sure I follow